src/main/java/com/sulphate/chatcolor2/listeners/ChatListener.java

```diff
@@ -23,78 +19,10 @@ public void onEvent(AsyncPlayerChatEvent e) {
 
         String uuid = e.getPlayer().getUniqueId().toString();
         checkDefault(uuid);
-
-        boolean override = (boolean) MainClass.getUtils().getSetting("color-override");
-        String color = MainClass.getUtils().getColor(uuid);
-
-        if (e.getMessage().contains("&")) {
-            String colourised = CC2Utils.colourise(e.getMessage());
-
-            if (override) {
-                e.setMessage(ChatColor.stripColor(colourised)); // Gets rid of all colour.
-            }
-            else {
-                // If not overriding, then colourise the message and stop here.
-                e.setMessage(colourised);
-                return;
-            }
-        }
-
-        if (color.contains("rainbow")) {
-            if (e.getMessage().contains("&")) {
-                // If there is color symbols, we don't want to put 'rainbow' at the start.
-                color = "";
-            }
-            else {
-                String rseq = (String) MainClass.getUtils().getSetting("rainbow-sequence");
-
-                if (!verifyRainbowSequence(rseq)) {
-                    MainClass.getUtils().setSetting("rainbow-sequence", "abcde");
-                    rseq = "abcde";
-                }
-
-                String mods = color.replace("rainbow", "");
-                char[] colors = rseq.toCharArray();
-                char[] msgchars = e.getMessage().toCharArray();
-                int currentColorIndex = 0;
-
-                StringBuilder sb = new StringBuilder();
-                for (int i = 0; i < msgchars.length; i++) {
-                    if (currentColorIndex == colors.length) {
-                        currentColorIndex = 0;
-                    }
-                    if (msgchars[i] == ' ') {
-                        sb.append(" ");
-                    }
-                    else {
-                        sb.append('&').append(colors[currentColorIndex]).append(mods).append(msgchars[i]);
-                        currentColorIndex++;
-                    }
-                }
-
-                String end = CC2Utils.colourise(sb.toString());
-                e.setMessage(end);
-                return;
-            }
-        }
-
-        e.setMessage(CC2Utils.colourise(color) + e.getMessage());
-    }
-
-    public boolean verifyRainbowSequence(String seq) {
-
-        boolean verify = true;
-        List<String> cols = Arrays.asList("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f");
-        String[] chars = seq.split("");
-        for (String s : chars) {
-            if (!cols.contains(s)) {
-                verify = false;
-            }
-        }
-        return verify;
+        e.setMessage(CC2Utils.colouriseMessage(MainClass.getUtils().getColor(uuid), e.getMessage(), true));
     }
 
-    public void checkDefault(String uuid) {
+    private static void checkDefault(String uuid) {
```

Comment:
Again just curious, I've never used a private static before.

Reply:
The reasons for this are analogous to the ones I stated above :)